compiler/hash-typecheck/src/ops/simplify.rs

```diff
@@ -371,18 +407,21 @@ impl<'tc> Simplifier<'tc> {
             }
             // Nominals:
             Level1Term::NominalDef(nominal_def_id) => {
-                let reader = self.reader();
-                let nominal_def = reader.get_nominal_def(*nominal_def_id);
-                match nominal_def {
-                    NominalDef::Struct(_struct_def) => {
+                match self.reader().get_nominal_def(*nominal_def_id) {
+                    NominalDef::Struct(_) => {
                         // Struct type access is not valid.
                         does_not_support_access(access_term)
                     }
                     NominalDef::Enum(enum_def) => {
                         // Enum type access results in the runtime value of the variant
                         // (namespace operation).
                         does_not_support_prop_access(access_term)?;
-                        match enum_def.variants.get(&access_term.name) {
+
+                        let Field::Named(name) = access_term.name else {
+                            unreachable!()
+                        };
```

Comment:
Shouldn't this be a user error?

Reply:
It's handled in `does_not_support_prop_access` in the call above.

Reply:
Maybe that function could return the `name` to avoid having to write unreachable.